lib/select.js

```diff
@@ -820,6 +820,10 @@ global.select = factory();
                     i = this.store.getItemsFilteredByActive(),
                     n = this._canAddItem(i, t);
 
+                v.triggerEvent(this.passedElement, "changeQuery", {
+                  value: t
+                })
```

Comment:
Is het niet veiliger om zelf een (query changed of search) event te versturen?

```JS
const event = 'changeQuery';
this._wrapperElement.addEventListener('keyup', () => this.dispatchEvent(new CustomEvent(event));
```

In de huidige code gebeurt niet meer dan dat, waardoor een wijziging (of logica) bij Choices.js eigenlijk niet nodig is.

```JS
this.containerOuter.addEventListener("keyup", this._onKeyUp);
```

Reply:
Maar je moet vanuit de applicatie ook wel aan de ingevulde zoekterm geraken. 
Die search input is volgens mij 'intern' aan choices.js. Zoals het nu gemaakt is, is dat simpel door te geven via de event, maar als je vanuit vl-select een event gaat throwen moet je de "abstractie" van choices.js doorbreken en de search input zelf gaan selecteren om de value er uit te halen.
Daarnaast is luisteren op een keyup op de container volgens mij ook niet helemaal 'correct'. De container != de search input. Het zal misschien wel werken (ik heb het nog niet getest, maar jij misschien wel?), maar semantisch is dat natuurlijk een beetje vreemd.

Het lijkt me in ieder geval eerder een concern van choices.js, dus als het puur is omdat we lievers niets aan choices.js code willen wijzigen, lijkt me dat geen goede reden. Zeker omdat choices.js open source is. 
Maar da's enkel mijn mening en jij bent de UIG man he. Zolang we onze event krijgen ben ik content :)

Reply:
Ik ben er de rest van de week waarschijnlijk niet meer, maar @pascalriquier gaat dit verder opvolgen.

Als we vanuit vl-select zouden luisteren op de keyUp van de search input zelf zou ik het nog ok vinden. Het nadeel dan is dat je de abstractie van choices.js doorbreekt omdat je rechtstreeks met de search input moet interageren vanuit vl-select. Ik zag het liever dichter bij de bron omdat je daar wat meer controle hebt over alles en omdat je niet doorheen die abstractie moet gaan, maar als jij het liever in vl-select ziet snap ik dat ook wel en daar kan ik me bij neerleggen.

Echter, een algemene keyup listener van heel de wrapper in vl-select lijkt me niet correct. Je focus kan namelijk ook op iets anders liggen dan de search input en als je dan op je toetsenbord zit te rammen ga je onnodig/incorrecte keyup events genereren.

Reply:
Het is inderdaad geen goed idee om op het `_wrapperElement` de 'keyup' op te vangen, dat zou dan specifieker moeten.

Ik ben alleen bezorgd dat jullie potentieel bij toekomstige updates problemen kunnen ondervinden die via de webcomponent testen niet naar boven komen. In dit geval bestaat de kans dat bij een volgende huisstijl upgrade de lib overschreven of vervangen wordt aangezien de testen groen zijn zonder de event propagatie.

Ik ben voorstander om de code wijzigingen zo dicht mogelijk bij de bron uit te voeren. Let wel op dat de [PR bij Choices](https://github.com/jshjohnson/Choices/pull/945) falende testen en lint problemen heeft. Eventueel dat nog even bekijken in de hoop dat de PR ooit goedgekeurd wordt. Geen probleem om in de tussentijd af te wijken via onze lib, maar dan zou ik liefst zekerheid inbouwen via een (bij voorkeur) unit test.

Reply:
De choice.js testen waren lokaal groen, ik heb de indruk dat hun CI omgeving de browsers gewoon niet opgestart kreeg.

Reply:
Ik heb een e2e test toegevoegd om regressies te voorkomen.